src/test/ui/const-generics/cannot-infer-type-for-const-param.rs

```diff
@@ -0,0 +1,11 @@
+#![feature(const_generics)]
+//~^ WARN the feature `const_generics` is incomplete and may cause the compiler to crash
+
+// We should probably be able to infer the types here. However, this test is checking that we don't
+// get an ICE in this case. It may be modified later to not be an error.
+
+struct Foo<const NUM_BYTES: usize>(pub [u8; NUM_BYTES]);
+
+fn main() {
+    let _ = Foo::<3>([1, 2, 3]); //~ ERROR type annotations needed
```

Comment:
What happens if this is replaced with `3usize`?

Reply:
Unfortunately, it ICEs:
```
src/librustc_codegen_llvm/context.rs:862: failed to get layout for `[type error]`: the type `[type error]` has an unknown layout
```
I think this is probably a result of the [existing issues](https://github.com/rust-lang/rust/issues/60619) with const generics and arrays. I'll add the updated test as a new issue.